src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/MockGoogleDataprocDAO.scala

```diff
@@ -1,20 +1,28 @@
 package org.broadinstitute.dsde.workbench.leonardo.dao
 
+import java.io.File
 import java.util.UUID
 
-import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterRequest, ClusterResponse}
+import org.broadinstitute.dsde.workbench.leonardo.config.DataprocConfig
+import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.GoogleProject
+import org.broadinstitute.dsde.workbench.leonardo.model._
 
 import scala.collection.concurrent.TrieMap
 import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
 import scala.concurrent.{ExecutionContext, Future}
 
-class MockGoogleDataprocDAO extends DataprocDAO {
+class MockGoogleDataprocDAO(protected val dataprocConfig: DataprocConfig) extends DataprocDAO {
+
+  val clusters: mutable.Map[String, Cluster] = new TrieMap()  // Cluster Name and Cluster
+  val firewallRules: mutable.Map[GoogleProject, String] = new TrieMap()  // Google Project and Rule Name
+  val buckets: mutable.ArrayBuffer[String] = new ArrayBuffer() // Array of bucket names - not keeping track of google projects since it's all in leo's project
+  val bucketObjects: mutable.ArrayBuffer[(String, String)] = new ArrayBuffer()  // Bucket Name and File Name
 
-  private val clusters: mutable.Map[String, Cluster] = new TrieMap()
 
   private def googleID = UUID.randomUUID().toString
 
-  override def createCluster(googleProject: String, clusterName: String, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[ClusterResponse] = {
+  override def createCluster(googleProject: String, clusterName: String, clusterRequest: ClusterRequest, bucketName: String)(implicit executionContext: ExecutionContext): Future[ClusterResponse] = {
```

Comment:
`bucketName` unused?

Reply:
`bucketName` in `createCluster` is passed to `buildCluster`, which uses it to create the `GoogleBucketUri` of the cluster init script and passes it to `setExecutableFile` on the cluster.

In the mock DAO, since we're not creating an actual dataproc cluster, I didn't do anything with the `bucketName`. I could have the mock `createCluster` make the uri and then save it

Reply:
Should I save the `bucketName` somehow? It's already saved in the mock by `createBucket`

Reply:
oh right, this is the mock inheriting the signature from the proper DAO. this is fine, disregard